byterun/debugger.c

```diff
@@ -194,11 +195,16 @@ void caml_debugger_init(void)
     /* Unix domain */
     sock_domain = PF_UNIX;
     sock_addr.s_unix.sun_family = AF_UNIX;
+    a_len = strlen(address);
+    if (a_len >= sizeof(sock_addr.s_unix.sun_path)) {
+      caml_fatal_error("Debug socket path length exceeds maximum permitted length");
+    }
     strncpy(sock_addr.s_unix.sun_path, address,
-            sizeof(sock_addr.s_unix.sun_path));
+            sizeof(sock_addr.s_unix.sun_path) - 1);
+    sock_addr.s_unix.sun_path[sizeof(sock_addr.s_unix.sun_path) - 1] = '\0';
     sock_addr_len =
```

Comment:
Many authors just `memset` the whole `sockaddr_un` structure to zero before initializing the fields of interest.  I don't know if it's just superstition or a more robust way of having the path zero-terminated.

Reply:
Agreed -- the `sock_addr` value here should be initialised to 0 already as its in the BSS, so this zero termination is to be doubly sure (and in case the debugger can somehow be initialised twice, although that should never happen in normal use).